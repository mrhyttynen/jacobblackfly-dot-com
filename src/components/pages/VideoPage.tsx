import { useEffect, useState } from "react";
import { VerticalFlexBox } from "../../styles/styles";

export const VideoPage: React.FC = () => {

  // get data from Youtube API
  const [subscriberCount, setSubscriberCount] = useState(null);
  const [viewCount, setViewCount] = useState(null);
  useEffect(() => {
    const fetchSubscribers = async () => {
      try {
        // youtube API key (prd) has been configured on GCP to only accept requests from the website
        const response = await fetch(
          `https://www.googleapis.com/youtube/v3/channels?part=statistics&id=${process.env.REACT_APP_YOUTUBE_CHANNEL_ID}&key=${process.env.REACT_APP_YOUTUBE_API_KEY}`
        );
        const data = await response.json();
        if (data.items && data.items.length > 0) {
          setSubscriberCount(data.items[0].statistics.subscriberCount);
          setViewCount(data.items[0].statistics.viewCount);
        } else {
          console.warn('API returned no data for some reason');
        }
      } catch (error) {
        console.error(`Youtube API request failed! : ${error}`);
      }
    };

    fetchSubscribers();
  }, []);
  
  // return component
  return (
    <VerticalFlexBox>
      <h1>Subscribers: <b style={{textDecoration:"underline", textDecorationThickness: "2px"}}>{subscriberCount}</b> - can we get to 100?</h1>
      <h1>Total views: <b style={{textDecoration:"underline", textDecorationThickness: "2px"}}>{viewCount}</b></h1>
      <p>Refresh page to update</p>
      <iframe 
        className="w-[90vw] max-w-[1000px] h-[400px] sm:h-[500px] md:h-[600px] lg:h-[700px]"
        src="https://www.youtube.com/embed/drN7_hmVLh4?si=T66AoBorZzRFHMCS" 
        title="YouTube video player" 
        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" 
        referrerPolicy="strict-origin-when-cross-origin" 
        allowFullScreen
      >
      </iframe>
      <iframe 
        className="w-[90vw] max-w-[1000px] h-[400px] sm:h-[500px] md:h-[600px] lg:h-[700px]"
        src="https://www.youtube.com/embed/BMzRTILK1Aw?si=26gjodRhqgoIoy_u" 
        title="YouTube video player"
        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" 
        referrerPolicy="strict-origin-when-cross-origin" 
        allowFullScreen
      >
      </iframe>
    </VerticalFlexBox>
  );
};


