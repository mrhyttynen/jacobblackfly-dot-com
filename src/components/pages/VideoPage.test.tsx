import React from "react";
import { render, screen } from "@testing-library/react";
import { VideoPage } from "./VideoPage";
import renderer from "react-test-renderer";

describe("VideoPage", () => {
  it("renders page that matches snapshot", () => {
    const tree = renderer.create(<VideoPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
