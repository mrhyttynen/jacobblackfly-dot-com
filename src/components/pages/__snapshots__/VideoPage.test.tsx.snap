// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPage renders page that matches snapshot 1`] = `
<div
  className="sc-bcXHqe gkJjOJ"
>
  <h1>
    Subscribers:
     
    <b
      style={
        Object {
          "textDecoration": "underline",
          "textDecorationThickness": "2px",
        }
      }
    />
     
    - can we get to 100?
  </h1>
  <h1>
    Total views:
     
    <b
      style={
        Object {
          "textDecoration": "underline",
          "textDecorationThickness": "2px",
        }
      }
    />
  </h1>
  <p>
    Refresh page to update
  </p>
  <iframe
    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
    allowFullScreen={true}
    className="w-[90vw] max-w-[1000px] h-[400px] sm:h-[500px] md:h-[600px] lg:h-[700px]"
    referrerPolicy="strict-origin-when-cross-origin"
    src="https://www.youtube.com/embed/drN7_hmVLh4?si=T66AoBorZzRFHMCS"
    title="YouTube video player"
  />
  <iframe
    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
    allowFullScreen={true}
    className="w-[90vw] max-w-[1000px] h-[400px] sm:h-[500px] md:h-[600px] lg:h-[700px]"
    referrerPolicy="strict-origin-when-cross-origin"
    src="https://www.youtube.com/embed/BMzRTILK1Aw?si=26gjodRhqgoIoy_u"
    title="YouTube video player"
  />
</div>
`;
